src/main/java/core/data/medImage/ImageSeries.java

```diff
@@ -10,17 +10,21 @@
 public class ImageSeries extends Data implements Iterable<MedImage>, Cloneable {
 
     //TODO: manage access to the fields after image model is chosen
-    public List<MedImageTag> tags;
+    public List<MedImageAttribute> attributes;
     public List<MedImage> images;
 
-    public ImageSeries(List<MedImageTag> tags, List<MedImage> images) {
-        this.tags = tags;
+    public ImageSeries(List<MedImageAttribute> attributes, List<MedImage> images) {
+
+        this.attributes = new ArrayList<>();
+
+        if(attributes != null)
+            this.attributes.addAll(attributes);
+
         this.images = images;
     }
 
-    public MedImageTag findTag(UUID tagId) {
-        //TODO: (avlomakin) read more about java streams (LINQ C#)
-        Optional<MedImageTag> result = tags.stream().filter(x -> x.id.compareTo(tagId) == 0).findFirst();
+    public MedImageAttribute findTag(String tag) {
+        Optional<MedImageAttribute> result = attributes.stream().filter(x -> x.tag.equals(tag)).findFirst();
```

Comment:
NIT: How about using indexOf() on List instead? It returns first occurrence or -1 otherwise. Seems like it may be more readible

Reply:
index of what? Find by attribute tag, not by attribute itself

Reply:
Sorry, missed the part that a field needs to be accessed first. NVM